packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -235,11 +236,7 @@ export const Datagrid: React.ForwardRefExoticComponent<
      * the Datagrid displays the empty component.
      */
     if (data == null || data.length === 0 || total === 0) {
-        if (empty) {
-            return empty;
-        }
-
-        return null;
+        return empty ?? <ListNoResults resource={resource} />;
```

Comment:
If `empty` is `null` or `false`, it shouldn't return a default component. Only when it is `undefined`